### DISCUSSION TITLE
@unique cant work in mongodb

### DISCUSSION THREAD

**Discussion by wintsa123** [prisma/prisma#22737] [Q&A] [answered]
### Question

```prisma
model user {
  id       String  @id @default(auto()) @map("_id") @db.ObjectId
  phone    String  @unique
  name     String?  @unique
  password String
  role     String?
  avatar   String?
}
```
i tried to create same name data,i found it successfully! @unique cant work

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @wintsa123 👋 

Thank you for raising this question.

> i tried to create same name data,i found it successfully! @unique cant work

Can you please clarify the issue you are facing? Is the unique constraint not enforced when you try to create the same record multiple time?

**Reply by wintsa123** [reply]
Yes, I added @unique , but the unique constraint is not enforced when inserting data using Code Tim. I am using mongodb.

**Comment by janpio** [comment, marked as answer]
Did you run `prisma db push` after adding the `@unique` to your Prisma schema, so that the unique constraint/index would actually be created on your collection?

**Reply by wintsa123** [reply]
I didn't. I'm going to try.

**Reply by wintsa123** [reply]
good answer！！！
